### PR TITLE
Internal API returns unified errors

### PR DESCRIPTION
**Pull request by kmakiela**
* Added error definitions in Service behaviour
* Added modules handling errors in both services

### CODE REVIEW

File: lib/mongoose_push/service/apns/error_handler.ex
Comment (rslota) [inline lib/mongoose_push/service/apns/error_handler.ex]: How about we merge `unavailable` and `service_internal`? Those seem to be roughly the same from MongoosePush perspective.

File: lib/mongoose_push/service/fcm/error_handler.ex
Comment (rslota) [inline lib/mongoose_push/service/fcm/error_handler.ex]: How about we merge `apns_cert` with `auth`? Those look the same for me.

File: lib/mongoose_push.ex
Comment (rslota) [inline lib/mongoose_push.ex:106]: Since we know we return a tuple with 2 atoms, let's match on it and report both values in the log message separately, e.g.:
```suggestion
    {category, service_error} = reason
    Logger.warn(~s"Unable to complete push request due to service error: #{service_error} (category: #{category})")
```

**Comment by kmakiela** [comment]
PR should be ready to merge after rebasing to master after #113 is merged. 
* Added V2 error codes translation to HTTP2

File: lib/mongoose_push/service.ex
Comment (rslota) [inline lib/mongoose_push/service.ex]: Why do we need this in the behavior? This seems to be called only internally.

File: lib/mongoose_push/service/apns/error_handler.ex
Comment (rslota) [inline lib/mongoose_push/service/apns/error_handler.ex]: How does this differ from `BadDeviceToken`?

File: lib/mongoose_push/service/apns/error_handler.ex
Comment (rslota) [inline lib/mongoose_push/service/apns/error_handler.ex]: Let's rename `unregistered` to `invalid_device_token` and group all errors that relate to invalid device token.

File: lib/mongoose_push/service/apns/error_handler.ex
Comment (rslota) [inline lib/mongoose_push/service/apns/error_handler.ex]: This seems to be more for `internal_config` - this is not based on user input.

**Review by rslota — CHANGES_REQUESTED** [review]

**Review by rslota — APPROVED** [review]

**Pull request merged**
